apps/api/scripts/generate-secrets.ts

```diff
@@ -39,32 +39,35 @@ async function generateSecrets(): Promise<void> {
   const rootEncryptionKey = generateSecureSecret();
 
   // Replace placeholder values
-  const placeholderPatterns = [/ROOT_ENCRYPTION_KEY=.*$/m];
-
-  const replacements = [`ROOT_ENCRYPTION_KEY=${rootEncryptionKey}`];
+  const placeholderPatterns = [
+    {
+      pattern: /ROOT_ENCRYPTION_KEY=.*$/m,
+      replacement: `ROOT_ENCRYPTION_KEY=${rootEncryptionKey}`,
+    },
+  ];
```

Comment:
If we model this differently, we remove the possibility of undefined values when iterating through one array and pulling values from another array by index.